src/figvalidationservice/src/main/java/ca/bc/gov/open/pssg/rsbc/dps/figvalidationservice/ValidateApplicantForSharingController.java

```diff
@@ -24,21 +28,35 @@
 @RestController
 public class ValidateApplicantForSharingController {
 
+	private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
 	@Autowired
 	private FigaroValidationImpl figservice; // connection to ORDS client.
 
-	@RequestMapping(value = "/validateApplicantForSharing", produces = {
-			"application/xml" }, method = RequestMethod.GET)
-	@ApiOperation(value = "Validate Applicant For Sharing", notes = "", response = ValidateApplicantForSharingResponse.class, tags={ "Figaro Validation Services"})
-  	@ApiResponses(value = {@ApiResponse(code = 200, message = "Successful operation", response = ValidateApplicantForSharingResponse.class) })
-    
+	@RequestMapping(value = "/validateApplicantForSharing", produces = {"application/xml" }, method = RequestMethod.GET)
+	
+	@ApiOperation(value = "Validate Applicant For Sharing", notes = "", response = ValidateApplicantForSharingResponse.class, tags = {
+			"Figaro Validation Services" })
+	@ApiResponses(value = {
+			@ApiResponse(code = 200, message = "Successful operation", response = ValidateApplicantForSharingResponse.class) })
+
 	public ValidateApplicantForSharingResponse validateApplicantForSharingResponse(
-			@ApiParam(value = "applPartyId", required = false) @RequestParam(value = "applPartyId", defaultValue = "0",required = false) String applPartyId,
-			@ApiParam(value = "jurisdictionType", required = false) @RequestParam(value = "jurisdictionType", defaultValue = "",required = false) String jurisdictionType)
+			@ApiParam(value = "applPartyId", required = false) @RequestParam(value = "applPartyId", defaultValue = "0", required = false) String applPartyId,
+			@ApiParam(value = "jurisdictionType", required = false) @RequestParam(value = "jurisdictionType", defaultValue = "", required = false) String jurisdictionType)
 			throws FigaroValidationServiceException {
 
-		return figservice
-				.validateApplicantForSharing(new ValidateApplicantForSharingRequest(applPartyId, jurisdictionType));
+		try {
+
+			return figservice
+					.validateApplicantForSharing(new ValidateApplicantForSharingRequest(applPartyId, jurisdictionType));
+
+		} catch (FigaroValidationServiceException ex) {
+			logger.error("Exception caught as ValidateApplicantService : " + ex.getMessage());
+			ex.printStackTrace();
```

Comment:
do we need to print the stack trace if we're logging it already?

Reply:
Yes. This is common practice. At least within our group at the AG. It's allows the exception to be caught in the situation when logging has not been set up properly which BTW happens a lot although mostly in older applications.

Reply:
thanks